Makefile

```diff
@@ -233,7 +240,7 @@ emulation-setup:
 
 
 .PHONY: emulation-check
-emulation-check: emulation-setup
+emulation-check:
```

Comment:
I understand the motivation behind removing the implicit setup step, but in the spirit of "help new people to do the right thing automatically", could you add a check to the target here, probably something like:

`if [ ! -x tools/qemu/riscv32-softmmu/qemu-system-riscv32 ]; echo "Error: Running emulation checks requires qemu. Please run 'make emulation-setup' to prepare. This will take several minutes, but only needs to be run once" && exit 1; fi`

---

Generally, the Tock build system tries pretty hard to avoid the idiom of "the command you tried to run failed, please type this other command and now try again", and instead simply does the work for you.

While I am sympathetic to the long build of qemu, it's also a one-off cost. I might be more inclined to just add a print to the `emulation-setup` rule that says something like, "First run requires building qemu. This will take a few minutes. You can cancel and run `make ci-no-qemu` to skip for now".

Reply:
Shouldn't `emulation-check` always depend on setup? How can it do the check without the setup? I guess it doesn't matter to me if it is a check in the makefile or a target dependency, but since we don't want to run it on every make call it kind of has to be a target dependency.

Reply:
> Generally, the Tock build system tries pretty hard to avoid the idiom of "the command you tried to run failed, please type this other command and now try again", and instead simply does the work for you.

On the other hand, I would argue that the current setup doesn't try that hard and fails pretty quickly without the necessary (Debian) packages.

```
$ make emulation-check 
Cloning into 'qemu'...
remote: Enumerating objects: 19007, done.
remote: Counting objects: 100% (19007/19007), done.
remote: Compressing objects: 100% (6200/6200), done.
remote: Total 482048 (delta 14287), reused 16209 (delta 12647), pack-reused 463041
Receiving objects: 100% (482048/482048), 195.57 MiB | 27.38 MiB/s, done.
Resolving deltas: 100% (394231/394231), done.
Checking connectivity... done.

ERROR: pkg-config binary 'pkg-config' not found

Makefile:219: recipe for target 'emulation-setup' failed
make: *** [emulation-setup] Error 1
```

And installing the required dependencies will likely require `root` privileges, which will anyway require manual user action beyond Tock's Makefile.

---

Could you add a check to the target here, probably something like:

`if [ ! -x tools/qemu/riscv32-softmmu/qemu-system-riscv32 ]; echo "Error: Running emulation checks requires qemu. Please run 'make emulation-setup' to prepare. This will take several minutes, but only needs to be run once" && exit 1; fi`

I'd prefer someone who has actually used the qemu rules to add that.

But in any case, yes the `emulation-check` rule should check for required dependencies and ask the user to install them before downloading 330MB of qemu history. By the way, can't the rule use `git clone --depth 1` to avoid cloning the whole history of qemu?

Reply:
Also, the next error is even less clear:

```bash
$ sudo apt install pkg-config
...
$ make emulation-check 
make[1]: *** [config-host.mak] Error 1
Makefile:219: recipe for target 'emulation-setup' failed
make: *** [emulation-setup] Error 2
```

Reply:
We could stop piping the output of make to `/dev/null`. That is why you don't have a clear error.